client/src/components/insert/Container.tsx

```diff
@@ -98,13 +106,36 @@ const InsertContainer: FC<InsertContentProps> = ({
     };
   }, [insertStatus]);
 
+  const checkUploadFileValidation = (fieldNamesLength: number): boolean => {
+    return schemaOptions.length === fieldNamesLength;
+  };
+
+  const previewData = useMemo(() => {
+    const end = isContainFieldNames ? 5 : 4;
+    return csvData.slice(0, end);
+  }, [csvData, isContainFieldNames]);
+
+  const handleUploadedData = (csv: string) => {
+    const { data } = parse(csv);
+    const uploadFieldNamesLength = (data as string[])[0].length;
+    const validation = checkUploadFileValidation(uploadFieldNamesLength);
+    if (!validation) {
+      // open snackbar
+      openSnackBar(insertTrans('uploadFieldNamesLenWarning'), 'error');
+      // reset filename
+      setFileName('');
+      return;
+    }
+    setCsvData(data);
+  };
+
   const handleInsertData = () => {
     // mock status change
     setInsertStauts(InsertStatusEnum.loading);
     handleInsert();
     setTimeout(() => {
       setInsertStauts(InsertStatusEnum.success);
-    }, 500);
+    }, 1000);
```

Comment:
why 1000?

Reply:
There use `setTimout` just to mock async event, will be deleted later.

Reply:
ok. cool